File: c++/1260.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;
queue<int> q;
vector<int> v[1001];
bool visit_b[1001];
bool visit_d[1001];

void dfs(int start){
    if(visit_d[start]==true){
        return;
    }
    visit_d[start] = true;
    cout << start << " ";
    for(int i=0 ; i<v[start].size() ; i++){
        int next = v[start][i];
        dfs(next);
    }
}


void bfs(int start){
    q.push(start);
    visit_b[start]=true;

    while(!q.empty()){
        cout << q.front() << " ";
        int num = q.front();
        q.pop();
        for(int i=0 ; i<v[num].size() ; i++){
            int temp = v[num][i];
            if(!visit_b[temp]){
                q.push(temp);
                visit_b[temp] = true;
            }
        }
    }
}


int N,M,V;
int a,b;
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    cin >> N >> M >> V;
    for(int i=0 ; i<M ; i++){
        cin >> a >> b;
        v[a].push_back(b);
        v[b].push_back(a);
    }
    for(int i=1 ; i<=N ; i++){
        sort(v[i].begin(),v[i].end());
    }
    dfs(V);
    cout << "\n";
    bfs(V);

    return 0;
}

File: python/1260.py
N, M, start = map(int, input().split())

relation = [[0 for i in range(N+1)] for j in range(N+1)]

visit = [0 for k in range(N+1)]
visit2 = [0 for l in range(N+1)]

a_dfs = []
a_bfs = []

for i in range(M):
    x, y = map(int, input().split())
    relation[x][y] = 1
    relation[y][x] = 1


def dfs(st):
    if visit[st] == 1:
        return 0
    visit[st] = 1
    a_dfs.append(st)
    for i in range(1, N+1):
        if visit[i] == 0 and relation[st][i] == 1:
            dfs(i)


def bfs(st):
    queue = []
    queue.insert(0, st)
    visit2[st] = 1
    a_bfs.append(st)
    while len(queue) > 0:
        now = queue[-1]
        queue.pop()
        for i in range(1, N+1):
            if visit2[i] == 0 and relation[now][i] == 1:
                queue.insert(0, i)
                a_bfs.append(i)
                visit2[i] = 1


dfs(start)
bfs(start)

for i in a_dfs:
    print(i, end=' ')

print()

for j in a_bfs:
    print(j, end=' ')
